src/client/gui/gui_cmd.cpp

```diff
@@ -34,6 +34,65 @@ namespace cmd = multipass::cmd;
 using RpcMethod = mp::Rpc::Stub;
 using namespace std::chrono_literals;
 
+namespace
+{
+auto set_title_string_for(const std::string& text, const mp::InstanceStatus& state)
+{
+    return QString::fromStdString(fmt::format("{}{}", text,
+                                              (state.status() != mp::InstanceStatus::STOPPED)
+                                                  ? fmt::format(" ({})", mp::format::status_string_for(state))
+                                                  : ""));
+}
+
+void set_input_state_for(QList<QAction*> actions, const mp::InstanceStatus& state)
+{
+    if (actions.isEmpty())
+        return;
+
+    // actions are in the following order:
+    // - Start action
+    // - Open Shell action
+    // - Stop action
```

Comment:
Hey, thanks for the review.  I actually went back and forth on where to put the comments between where I have it now and where you suggest.  Since you suggest putting it at the top of the function, then I'll do that :grin:

Reply:
OK thanks! I consider this approved then.

bors delegate+

Reply:
Ok, done.